Source/Core/UICommon/DiscordPresence.cpp

```diff
@@ -244,6 +249,18 @@ void UpdateDiscordPresence(int party_size, SecretType type, const std::string& s
       // Note: joining still works without partyMax
     }
   }
+#ifdef USE_RETRO_ACHIEVEMENTS
+  else if (Config::Get(Config::RA_ENABLED) && Config::Get(Config::RA_DISCORD_PRESENCE_ENABLED))
+  {
+    state_string = AchievementManager::GetInstance().GetRichPresence().data();
+    if (state_string.length() >= 128)
+    {
+      // 124 characters + 3 dots + null terminator - thanks to Stenzek for format
+      state_string.replace(124, 4, "...");
```

Comment:
Are you sure this is right? I think for strings > 128 this would insert the `...` in the middle of the string instead of replacing the end.

Reply:
Gah, I was thinking of it wrong too.

Probably should be something like: `state_string = state_string.substr(124) + std::string{"..."}`

Reply:
My understanding was that this would insert ...\0 into the middle of the string and thus cut it off when it copies to a cstring?

Reply:
Ah, yeah you're right, but that's really not obvious with how the code is written. But yeah fair enough.

Reply:
A more easily verifiable approach is this:
```suggestion
      state_string.resize(124);
      state_string += "...";
```
(not that it mattered here, but this won't have overhead since down-sizing string never re-allocates)

As a side note, `std::string`'s null terminator is not included in its `length()`, but the comment suggests a different assumption was made. Might be worth double-checking.